pkg/installer/dnsmasq/dnsmasq.go

```diff
@@ -0,0 +1,195 @@
+package dnsmasq
+
+import (
+	"bytes"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"text/template"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/coreos/go-semver/semver"
+	ign3types "github.com/coreos/ignition/v2/config/v3_2/types"
+	mcv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	"github.com/vincent-petithory/dataurl"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	configFileName    = "dnsmasq.conf"
+	unitFileName      = "dnsmasq.service"
+	prescriptFileName = "aro-dnsmasq-pre.sh"
+)
+
+func config(clusterDomain, apiIntIP, ingressIP string, gatewayDomains []string, gatewayPrivateEndpointIP string) ([]byte, error) {
+	t := template.Must(template.New(configFileName).Parse(configFile))
+	buf := &bytes.Buffer{}
+
+	err := t.ExecuteTemplate(buf, configFileName, &struct {
+		ClusterDomain            string
+		APIIntIP                 string
+		IngressIP                string
+		GatewayDomains           []string
+		GatewayPrivateEndpointIP string
+	}{
+		ClusterDomain:            clusterDomain,
+		APIIntIP:                 apiIntIP,
+		IngressIP:                ingressIP,
+		GatewayDomains:           gatewayDomains,
+		GatewayPrivateEndpointIP: gatewayPrivateEndpointIP,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return buf.Bytes(), nil
+}
+
+func service() (string, error) {
+	t := template.Must(template.New(unitFileName).Parse(unitFile))
+	buf := &bytes.Buffer{}
+
+	err := t.ExecuteTemplate(buf, unitFileName, nil)
+	if err != nil {
+		return "", err
+	}
+
+	return buf.String(), nil
+}
+
+func startpre() ([]byte, error) {
+	t := template.Must(template.New(prescriptFileName).Parse(preScriptFile))
+	buf := &bytes.Buffer{}
+
+	err := t.ExecuteTemplate(buf, prescriptFileName, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return buf.Bytes(), nil
+}
```

Comment:
These 3 (arguably 4) functions are the same except for the file name and content they operate on. Those could be parameters.